fix(button): increase contrast

caugner opened the pull request:
### Description

Increases the contrast between the background color with/without hover, all while ensuring a visible hover effect.

### Motivation

Resolve accessibility issues.

### Additional details

As of 120d223b1dfbc394c4f56ff4ffae004f4fa11e70:

### WCAG contrast ratios

| State                  | Light (main → branch)                                                                                                                                                                              | Dark (main → branch)                                                                                                                                                                                |
| ---------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| YES default (no hover) | [3.81:1](https://webaim.org/resources/contrastchecker/?fcolor=089141&bcolor=F7F7F8) 🔴 → [**7.74:1**](https://webaim.org/resources/contrastchecker/?fcolor=065B29&bcolor=F7F7F8) 🟢                  | [10.49:1](https://webaim.org/resources/contrastchecker/?fcolor=70EBA3&bcolor=212426) 🟢                                                                                                              |
| YES hover              | [3.66:1](https://webaim.org/resources/contrastchecker/?fcolor=089141&bcolor=EAF6EF) 🔴 → [**7.44:1**](https://webaim.org/resources/contrastchecker/?fcolor=065B29&bcolor=EAF6EF) 🟢                  | [9.56:1](https://webaim.org/resources/contrastchecker/?fcolor=70EBA3&bcolor=0C311C) 🟢 → [**5.54:1**](https://webaim.org/resources/contrastchecker/?fcolor=70EBA3&bcolor=065B29) 🟠                   |
| NO default (no hover)  | [5.02:1](https://webaim.org/resources/contrastchecker/?fcolor=D31232&bcolor=F7F7F8) 🟠                                                                                                              | [6.35:1](https://webaim.org/resources/contrastchecker/?fcolor=FC7D92&bcolor=212426) 🟠                                                                                                               |
| NO hover               | [4.53:1](https://webaim.org/resources/contrastchecker/?fcolor=D31232&bcolor=FFE5EA) 🟠 → [**5.67:1**](https://webaim.org/resources/contrastchecker/?fcolor=B7102C&bcolor=FFE5EA) 🟠                  | [6.24:1](https://webaim.org/resources/contrastchecker/?fcolor=FC7D92&bcolor=4B0B16) 🟠                                                                                                               |

Legend: 🔴 fails AA (<4.5:1) · 🟠 passes AA (≥4.5:1) · 🟢 passes AAA (≥7:1) — thresholds for normal text.


### Screenshots

| | [Before](https://developer.mozilla.org/en-US/docs/Web/CSS#see_also:~:text=Help%20improve%20MDN) | [After](https://fred-pr666.review.mdn.allizom.net/en-US/docs/Web/CSS#:~:text=Help%20improve%20MDN) |
|--------|--------|--------|
| Light Yes | <img width="1536" height="550" alt="image" src="https://github.com/user-attachments/assets/72794359-b0e5-4c8e-853a-e62f51b138a6" /> | <img width="1536" height="550" alt="image" src="https://github.com/user-attachments/assets/5260025a-ca95-4d67-926a-deeaf61dfde0" /> |
| Light No | <img width="1536" height="550" alt="image" src="https://github.com/user-attachments/assets/f34f899f-120f-44ec-b2b9-bf97e2e1666c" /> | <img width="1536" height="550" alt="image" src="https://github.com/user-attachments/assets/eb41baa2-3361-4049-921c-4d5def41ddae" /> | 
| Dark Yes | <img width="1536" height="550" alt="image" src="https://github.com/user-attachments/assets/2f6ce954-2900-4796-afed-1405cc7af543" /> | <img width="1536" height="550" alt="image" src="https://github.com/user-attachments/assets/3815f44d-1a9f-4e69-a956-00fda5d4f76e" /> |
| Dark No | <img width="1536" height="550" alt="image" src="https://github.com/user-attachments/assets/88e9d224-ad11-44a8-8527-8bf2fd2f0b56" /> | <img width="1536" height="550" alt="image" src="https://github.com/user-attachments/assets/88ca6eff-adc9-4c6b-9cee-6223e0033f04" /> |

### Related issues and pull requests

Fixes https://github.com/mdn/fred/issues/653.

caugner committed:
fix(button): increase contrast for light positive action

Only secondary/plain, as primary uses white text color.

caugner committed:
fix(button): increase contrast for light negative action

OnkarRuikar commented:
Please update the plain button hover contrast in dark mode as well:
```diff
 /* Plain */
  &[data-variant="plain"] {
    &:hover {
--      background-color: light-dark(var(--color-gray-80), var(--color-gray-20));
++      background-color: light-dark(var(--color-gray-80), var(--color-gray-40));
    }
```
The current hover is almost not noticeable. In the following video, the `Clear` button has the above change:

https://github.com/user-attachments/assets/e2ae1f39-970a-403c-befb-33f7be7f1653

In the above video, the buttons are zoomed in, but in the normal case hover color change is not visible.

caugner committed:
chore(sandbox): add disabled buttons

caugner committed:
fix(button): increase dark neutral hover contrast

caugner committed:
fix(button): increase light primary hover contrast

OnkarRuikar commented:
A separate PR for hover contrast? There are other places that could use more hover contrast:

https://github.com/user-attachments/assets/095d707d-e480-4e47-895b-aa0fcf8b4656

caugner committed:
Merge branch 'main' into 653-increase-positive-button-contrast

caugner committed:
fix(button): increase light positive default contrast

Use `--color-green-20` as the default text color for `secondary` and
`plain` positive buttons in light mode, lifting contrast on
`--color-background-primary` from 3.81:1 (fails AA) to 7.74:1 (AAA).
With the default color now matching the previous hover color, the
hover `color` override is redundant and removed.

caugner committed:
fix(button): increase dark positive hover contrast

Use `--color-green-20` instead of `--color-green-10` as the dark-mode
hover background for `secondary` and `plain` positive buttons. The
previous `green-10` is darker than the page background (`gray-10`),
making the hover state effectively invisible (1.10:1 luminance
contrast vs page). `green-20` lifts that to 1.90:1 — still below the
WCAG 1.4.11 3:1 threshold but materially more visible. Text contrast
on the new hover background remains AA (5.54:1).

caugner committed:
chore(sandbox): rename inner `disabled` to `isDisabled`

Avoid shadowing the outer `disabled` array declared at the top of the
render function.

caugner committed:
chore(sandbox): skip disabled anchor section

`disabled` is a non-standard attribute on `<a>`, so rendering a separate
disabled-anchor section duplicates the non-disabled one visually while
labeling both as `<a>`.

LeoMcA reviewed (approved):
Nice!
